Dark Mode

mehmetsagir opened the pull request:
I was wondering how the Animate.style website would look in dark mode and I tried it. I liked it and wanted to share it with you. You can view the renewed website here.

- [animate.style](https://animatecss-ms.netlify.app/)
- Additionally, I added a back to top button.

WarenGonzaga commented:
@mehmetsagir thanks... I'll review this and I would like to improve your progress. Thank you for taking time to work on this as I'm bit busy for this update.

WarenGonzaga commented:
@mehmetsagir kindly fix the remaining conflicting issue on your PR . thank you

mehmetsagir commented:
@WarenGonzaga okay, conflicts fixed

eltonmesquita commented on docs/index.html:
```diff
@@ -51,6 +51,20 @@ <h2 class="callout-subtitle">Just-add-water CSS animations</h2>
       <section class="animation-list">
         <button class="callout-hideList button">Close list</button>
+
+        <a class="dark-mode-button-container">
```
You have to make these changes in the [`docsSource/template.html`](https://github.com/animate-css/animate.css/blob/main/docsSource/template.html) file.

You'll also need to compile the docs using `npm run docs`.

eltonmesquita commented on docs/main.mjs:
```diff
@@ -23,3 +29,37 @@ document.querySelectorAll('.copy-icon').forEach(icon => {
     })
 })
+
+backToTopBtn.addEventListener('click', (e) => {
```
Let's avoid adding lot's of things to this file. Create a new module file in the [modules](https://github.com/animate-css/animate.css/tree/main/docs/modules) folder and wrap it all in an exported function which will get imported and executed in this file. See the other files in this folder for examples.

eltonmesquita commented on docs/main.mjs:
```diff
@@ -23,3 +29,37 @@ document.querySelectorAll('.copy-icon').forEach(icon => {
+        localStorage.setItem('animate-css', 'dark');
+        moon.style.fill = '#e0e0e0';
+        sun.style.fill = '#e0e0e0';
+    }else{
```
please, format your code properly: `} else {`.

eltonmesquita commented on docs/main.mjs:
```diff
@@ -23,3 +29,37 @@ document.querySelectorAll('.copy-icon').forEach(icon => {
+    }
+})
+
+window.addEventListener("scroll", () => {
```
please add `{ passive: true }` as a third argument to the `addEventListener` function. [See more about it here](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#improving_scrolling_performance_with_passive_listeners).

eltonmesquita commented on docs/main.mjs:
```diff
@@ -23,3 +29,37 @@ document.querySelectorAll('.copy-icon').forEach(icon => {
+
+window.addEventListener("scroll", () => {
+  if (window.pageYOffset > 400) {
+    backToTopBtn.style.display = "grid";
```
Please switch all your double quotes `"` for single ones `'`.

eltonmesquita commented on docs/style.css:
```diff
@@ -616,3 +616,126 @@ blockquote {
+  $DARK-MODE
+\*-----------------------------------*/
+
+.night-light-toggle-button .night-light-checkbox {
```
please avoid descended selector, it complicates your CSS and couple HTML to CSS too much.

eltonmesquita commented on docs/style.css:
```diff
@@ -616,3 +616,126 @@ blockquote {
+.night-light-toggle-button .night-light-checkbox {
+  position: absolute;
+  visibility: hidden;
+}
```
Please, add a new line between declarations.

eltonmesquita commented on docs/style.css:
```diff
@@ -616,3 +616,126 @@ blockquote {
+  position: absolute;
+  visibility: hidden;
+}
+.night-light-checkbox:checked + .night-light-label .night-light-ball {
```
Do we need descendent selector here?

eltonmesquita commented on docs/style.css:
```diff
@@ -616,3 +616,126 @@ blockquote {
+}
+.night-light-checkbox:checked + .night-light-label .night-light-ball {
+  transform: translateX(28px);
+}
```
new line

eltonmesquita commented on docs/style.css:
```diff
@@ -616,3 +616,126 @@ blockquote {
+  width: 17px;
+}
+
+body.dark{
```
remove the tag selector `body` as it's unnecessary.

eltonmesquita commented on docs/style.css:
```diff
@@ -616,3 +616,126 @@ blockquote {
+  color: #e0e0e0;
+}
+
+body.dark .intro{
```
remove `body`

eltonmesquita commented on docs/style.css:
```diff
@@ -616,3 +616,126 @@ blockquote {
+  background: #111;
+}
+
+body.dark .animation-list,
```
remove `body`

eltonmesquita commented on docs/style.css:
```diff
@@ -616,3 +616,126 @@ blockquote {
+}
+
+body.dark .animation-list,
+body.dark .sidebar{
```
remove `body`

eltonmesquita commented on docs/style.css:
```diff
@@ -616,3 +616,126 @@ blockquote {
+  background: #15151D;
+}
+
+body.dark .button.button-animations,
```
remove `body`

eltonmesquita commented on docs/style.css:
```diff
@@ -616,3 +616,126 @@ blockquote {
+}
+
+body.dark .button.button-animations,
+body.dark .callout-hideList.button{
```
remove `body`

eltonmesquita commented on docs/style.css:
```diff
@@ -616,3 +616,126 @@ blockquote {
+  outline: none;
+}
+
+body.dark .copy-icon,
```
remove `body`

eltonmesquita commented on docs/style.css:
```diff
@@ -616,3 +616,126 @@ blockquote {
+}
+
+body.dark .copy-icon,
+body.dark .copy-icon:hover{
```
remove `body`

eltonmesquita commented on docs/style.css:
```diff
@@ -616,3 +616,126 @@ blockquote {
+  width: 13px;
+}
+
+.back-to-top .up-svg path{
```
`.back-to-top` is unnecessary, please remove it. Also, format your code properly. Always add a space before a `{`

WarenGonzaga replied:
Good catch!

eltonmesquita commented on docs/style.css:
```diff
@@ -616,3 +616,126 @@ blockquote {
+body.dark a[title="Documentation"],
+body.dark pre *{
+  color: #e0e0e0;
+}
```
new line

eltonmesquita reviewed (changes requested):
Sorry for the late review, but you know, RealLife™️. Thanks for the PR it's looking great!

I added a bunch of comments, but they should be simple to fix. Most of them are code standards related as we currently don't have one 😅,  I'll try to work on that later. Don't feel down for having so many comments when PRing in open source projects, this is fairly usual.

I noticed that you have a bunch of commits and they have unrelated titles. Can you try to fix that? You can ask for help if you need, but I think it'd be easier if you use a GUI like SourceTree to undo the commits and properly name them as new commits. You'll have to force push your work, but it shouldn't be a problem.

eltonmesquita commented:
If you need any help, feel free to ask. I might take a bit long but I'll gladly help!

WarenGonzaga reviewed (commented):
Looks good to me... can you provide a preview of it?

mehmetsagir commented:
Elbette, [demo-animate.style](https://animatecss-ms.netlify.app/) üzerinden inceleyebilirsiniz. If there are places I need to fix again, just specify.

WarenGonzaga commented:
Hi @mehmetsagir can you fit this slider button?
![image](https://user-images.githubusercontent.com/15052701/108043519-27bebf80-707c-11eb-9b9b-13d885f5df9e.png)

Looks like not fit and I saw spacing...

WarenGonzaga commented:
A good reference might like this...
![image](https://user-images.githubusercontent.com/15052701/108043785-7bc9a400-707c-11eb-9664-92f047154b54.png)

mehmetsagir commented:
Ok i will edit

WarenGonzaga commented:
Awesome @mehmetsagir let us know... and don't forget to prepare an update preview.

mehmetsagir commented:
I edited the dark mode button. I deleted the blanks and simplified the code a bit. 
@WarenGonzaga, you can view domo-animate.css here. I would appreciate it if you let me know if there are places I need to organize.

WarenGonzaga reviewed (approved):
Smooth! Nice update! @eltonmesquita kindly review this... for me its all good.

eltonmesquita commented on docs/modules/darkMode.mjs:
```diff
@@ -0,0 +1,16 @@
+const darkCheck = document.getElementById('night-light-checkbox');
+
+darkCheck.addEventListener('click', () => {
+  if (darkCheck.checked) {
+    document.body.classList.add('dark');
+    localStorage.setItem('animate-css', 'dark');
+  } else {
+    document.body.classList.remove('dark');
+    localStorage.removeItem('animate-css');
+  }
+})
+
+if (localStorage.getItem('animate-css')) {
+  document.body.className = 'dark';
+  darkCheck.checked = true;
+}
```
Could you please wrap this whole file in a function and export it just like the other modules?

eltonmesquita commented on docs/main.mjs:
```diff
@@ -2,6 +2,7 @@ import buildDocsIndex from './modules/buildDocsIndex.mjs';
 import playground from './modules/playground.mjs';
 import startAnimations from './modules/startAnimations.mjs';
 import toggleOnClick from './modules/toggle.mjs';
+import('./modules/darkMode.mjs');
```
Please use a named import and execute the function bellow.

eltonmesquita reviewed (changes requested):
Thanks a lot for all the effort, it's looking great! One thing that is not yet addressed is the git commit history. It's just too messy right now and I prefer to avoid merging it as it is right now. In my opinion, one single commit would be enough for this PR. Make sure to also give a short, yet descriptive, commit message.

It can be daunting to rewrite git history, but it is easier than it seems. [This article](https://www.atlassian.com/git/tutorials/rewriting-history) is a good intro into git history rewriting. You can also use a GUI as I mentioned before. Feel free to ask for help if you're having a hard time doing this.

mehmetsagir commented:
I fixed the commit messages of all the pages I was editing. I'm not sure if it's correct, can you review when appropriate? @eltonmesquita

eltonmesquita reviewed (approved):
Looks good! Thanks for all the effort 🎖️
